smartsim/_core/entrypoints/service.py

```diff
@@ -98,13 +116,30 @@ def execute(self) -> None:
         """The main event loop of a service host. Evaluates shutdown criteria and
         combines with a cooldown period to allow automatic service termination.
         Responsible for executing calls to subclass implementation of `_on_iteration`"""
-        self._on_start()
+
+        try:
+            self._on_start()
+        except Exception:
+            logger.exception("Unable to start service.")
+            return
 
         running = True
         cooldown_start: t.Optional[datetime.datetime] = None
 
         while running:
-            self._on_iteration()
+            try:
+                self._on_iteration()
+            except Exception:
+                running = False
+                logger.exception(
+                    "Failure in event loop resulted in service termination"
+                )
+
+            if self._health_check_frequency > 0:
```

Comment:
I suggest `>=` to allow users to specify `0` as an effective value.

Reply:
Allowed zero. Added tests to `test_service.py` to verify hc freq > 0 and hc freq == 0